urldeb33fd2fa_friedemann_distripedia_tgz-pJ8-org_json_TestJ8/src/wikipedia/Main.java

```diff
@@ -0,0 +1,221 @@
+package wikipedia;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URLConnection;
+import java.net.URLEncoder;
+import java.util.Arrays;
+import java.util.Properties;
+import java.util.logging.Handler;
+import java.util.logging.LogRecord;
+
+import websocket.common.Logger;
+import websocket.server.WebSocket;
+import websocket.server.WebSocketServer;
+import wikipedia.utils.URLUtils;
+
+public class Main
+{
+	final static PrintStream ps = System.out;
+
+	final static String wikipedia_url = "http://en.wikipedia.org/w/index.php";
+	final static String wikipedia_args = "?action=render&printable=yes&title=";
+
+	final static String article_file_path = "/wiki/";
+	final static int articles_max = 5000;
+	static String articles_counter_file = "/WEB-INF/properties/articles_counter.properties";
+	static Properties articles_prop;
+
+	/**
+	 * Get the article as XML from Wikipedia
+	 * 
+	 * @param article_title
+	 * @return
+	 * @throws IOException 
+	 */
+	private static String get_xmlFromWikipedia(String article_title) throws IOException
+	{
+		String encodedArticle = null;
+		
+		try
+		{
+			encodedArticle = URLEncoder.encode(article_title, "UTF-8");
+		} catch (UnsupportedEncodingException e)
+		{
+			e.printStackTrace();
+		}
+		
+		URLConnection conn = URLUtils.getURLConnection(wikipedia_url + wikipedia_args + encodedArticle);
+		
+
+		return URLUtils.getSource(conn);
+	}
+
+	/**
+	 * Save the XML file
+	 * 
+	 * @param file_name
+	 * @param content
+	 */
+	private static void save_xmlFile(String file_name, String content)
+	{
+		BufferedWriter bw = null;
+		try
+		{
+
+			bw = new BufferedWriter(new FileWriter(file_name));
+			bw.write(content);
+
+		} catch (IOException exc)
+		{
+			// Logger.getLogger (Main.class.getName ()).log (Level.SEVERE, null,
+			// exc);
+			exc.printStackTrace();
+		} finally
+		{
+
+			try
+			{
+				if (bw != null)
+					bw.close();
+			} catch (IOException e)
+			{
+				// TODO Auto-generated catch block
+				e.printStackTrace();
+			}
+		}
+	}
+
+	/**
+	 * Get total amount of articles on own server
+	 * 
+	 * @return
+	 */
+	private static int get_totalArticles()
+	{
+		try
+		{
+			FileInputStream propInFile = new FileInputStream(WebSocketServer.path
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *propInFile*. Execution paths that do not contain closure statements were detected. Ensure *propInFile* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).